core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/deploysemantics/PhysicalLink.scala

```diff
@@ -0,0 +1,126 @@
+package edu.uci.ics.amber.engine.architecture.deploysemantics
+
+import edu.uci.ics.amber.engine.architecture.sendsemantics.partitionings.{
+  BroadcastPartitioning,
+  HashBasedShufflePartitioning,
+  OneToOnePartitioning,
+  Partitioning,
+  RangeBasedShufflePartitioning,
+  RoundRobinPartitioning
+}
+import edu.uci.ics.amber.engine.common.AmberConfig.defaultBatchSize
+import edu.uci.ics.amber.engine.common.virtualidentity.{ActorVirtualIdentity, PhysicalLinkIdentity}
+import edu.uci.ics.texera.workflow.common.workflow.{
+  BroadcastPartition,
+  HashPartition,
+  PartitionInfo,
+  RangePartition,
+  SinglePartition,
+  UnknownPartition
+}
+
+object PhysicalLink {
+  def apply(
+      fromPhysicalOp: PhysicalOp,
+      fromPort: Int,
+      toPhysicalOp: PhysicalOp,
+      inputPort: Int
+  ): PhysicalLink = {
+    new PhysicalLink(
+      fromPhysicalOp,
+      fromPort,
+      toPhysicalOp,
+      inputPort,
+      Array()
+    )
+  }
+  def apply(
+      fromPhysicalOp: PhysicalOp,
+      fromPort: Int,
+      toPhysicalOp: PhysicalOp,
+      inputPort: Int,
+      partitionInfo: PartitionInfo
+  ): PhysicalLink = {
+    val partitionings: Array[(Partitioning, Array[ActorVirtualIdentity])] = partitionInfo match {
+      case HashPartition(hashColumnIndices) =>
+        fromPhysicalOp.identifiers
+          .map(_ =>
+            (
+              HashBasedShufflePartitioning(
+                defaultBatchSize,
```

Comment:
We only allow default `batchSize` now? Previously it was specifiable in `LinkStrategy`. If so maybe mention it in the PR description?

Reply:
it is the same as before, we used to initialize LinkStrategies with defaultBatchSize. Now as we merged LinkStrategy into PhysicalLink, this behavior is the same as before. In a future PR we will work on creating workflow execution config to capture those configurations.